File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> index;
        
        map<int, int> elementsSeen;

        for(int i = 0; i< nums.size(); i++){
            int difference = target - nums[i];
            if(elementsSeen.count(difference)){
                index.push_back(elementsSeen[difference]);
                index.push_back(i);
                return index;
            }
            elementsSeen.insert({nums[i], i});
        }
        return index;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen_map = dict()

        for i in range(len(nums)):
            dif = target - nums[i]

            if (dif in seen_map):
                return [seen_map[dif], i]
            
            seen_map[nums[i]] = i
         
